influence_benchmark/stats/preferences_per_iteration.py

```diff
@@ -50,16 +53,19 @@ def get_func_n_trajectories(
     top_n = (
         avg_rewards.groupby(["env_name", "initial_state_id"])
         .apply(
-            lambda x: x.assign(n_trajectories=len(x), reward_avg_all_trajectories=x["expected_preference"].mean()).pipe(
-                func, num_chosen_trajectories, "expected_preference"
-            )
+            lambda x: x.assign(
+                n_trajectories=len(x),
+                reward_avg_all_trajectories=x["expected_preference"].mean(),
+                influence_score_avg_all_trajectories=x["expected_influence_score"].mean(),
+            ).pipe(func, num_chosen_trajectories, "expected_preference")
         )
         .reset_index(drop=True)
     )
 
-    top_n = top_n.assign(reward_avg_selected_trajectories=top_n["expected_preference"]).drop(
-        columns=["expected_preference"]
-    )
+    top_n = top_n.assign(
+        reward_avg_selected_trajectories=top_n["expected_preference"],
+        influence_score_avg_selected_trajectories=top_n["expected_influence_score"],
+    ).drop(columns=["expected_preference", "expected_influence_score"])
 
     # Merge with original trajectories and select the longest for each group
```

Comment:
Where is the "select the longest" happening here? Is this comment still up to date?

Reply:
yes, .pipe(func, n_chosen_trajs, "traj_mean_rew") does this

Reply:
Yes sorry, ended up realizing the answer to this later

Reply:
This will be addressed separately